tools/spice/spice_utils.py

```diff
@@ -0,0 +1,159 @@
+    """
+
+    def __init__(self, kernel_path: Path, extensions: Optional[list[str]] = None):
+        self.kernel_path = kernel_path
+        if extensions is None:
+            self.extensions = [".bpc", ".bsp", ".ti", ".tf", ".tls", ".tsc"]
+        else:
+            self.extensions = extensions
+
+    def furnsh(self):
+        """Load a kernel file into the SPICE system."""
+        self.clear()
+        files_loaded = []
+
+        for filename in os.listdir(self.kernel_path):
+            if filename.endswith(tuple(self.extensions)):
+                filepath = os.path.join(self.kernel_path, filename)
+                spice.furnsh(filepath)
+                files_loaded.append(filename)
+            else:
+                logger.info(f"Invalid kernel_type extension: {filename}.")
+
+        return files_loaded
+
+    def clear(self):
+        """Clear the SPICE kernel pool."""
+        logger.info("Clearing SPICE kernel pool.")
+        spice.kclear()
+
+
+def ls_kernels(extensions=None) -> list:
+    """List furnished spice kernels, optionally filtered by specific extensions.
+
+    Parameters
+    ----------
+    extensions: list of str, optional
+        Extensions to filter the kernels. If None, list all kernels.
+
+    Returns
+    -------
+    : list
+        A list of kernel filenames.
+    """
+    count = spice.ktotal("ALL")
+    result = []
+
+    for i in range(count):
+        file, _, _, _ = spice.kdata(i, "ALL")
+        # Append the file if no specific extensions are provided
+        if extensions is None:
+            result.append(file)
+        # Check if the file ends with any of the specified extensions
+        elif any(file.endswith(ext) for ext in extensions):
+            result.append(file)
+
+    return result
+
+
+def ls_spice_constants() -> dict:
+    """List all constants in the Spice constant pool.
+
+    Returns
+    -------
+    : dict
+        Dictionary of kernel constants
+    """
+    try:
+        # retrieve names of kernel variables
+        # * means matches all variables
+        # 0 : starting from first match
+        # 1000 : num variable names retrieved in call
+        # 81 : max len of variable name
+        kervars = spice.gnpool("*", 0, 1000, 81)
+    except NotFoundError:  # Happens if there are no constants in the pool
+        return {}
+
+    result = {}
+    for kervar in sorted(kervars):
+        # retrieve data about a kernel variable
+        n, kernel_type = spice.dtpool(kervar)  # pylint: disable=W0632
+        # numerical data type
+        if kernel_type == "N":
+            values = spice.gdpool(kervar, 0, n)
+            result[kervar] = values
+        # character data type
+        elif kernel_type == "C":
+            values = spice.gcpool(kervar, 0, n, 81)
+            result[kervar] = values
+    return result
+
+
+def ls_attitude_coverage(custom_pattern=None) -> tuple:
+    """Process attitude kernels to extract and convert dates to UTC format.
```

Comment:
This might change. It was my first attempt at figuring out how we would select the most recent historical attitude kernel. I will put a TODO here in case we find that symlink can do that.